subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientHelper.java

```diff
@@ -222,6 +223,7 @@ private synchronized CloseableHttpClient getClient() {
         if (client == null) {
             HttpClientBuilder builder = HttpClientBuilder.create();
             new HttpClientConfigurer(settings).configure(builder);
+            builder.setRetryHandler(new DefaultHttpRequestRetryHandler(3, true));
```

Comment:
I think we could skip this part, haven't seen any major improvement in fetch operations after this change.